static_frame/core/util.py

```diff
@@ -288,66 +288,80 @@ class PairRight(Pair):
 
 #-------------------------------------------------------------------------------
 
-def mloc(array: np.ndarray) -> int:
-    '''Return the memory location of an array.
-    '''
-    return tp.cast(int, array.__array_interface__['data'][0])
-
-
-def immutable_filter(src_array: np.ndarray) -> np.ndarray:
-    '''Pass an immutable array; otherwise, return an immutable copy of the provided array.
-    '''
-    if src_array.flags.writeable:
-        dst_array = src_array.copy()
-        dst_array.flags.writeable = False
-        return dst_array
-    return src_array # keep it as is
-
-def name_filter(name: NameType) -> NameType:
-    '''
-    For name attributes on containers, only permit recursively hashable objects.
-    '''
-    try:
-        hash(name)
-    except TypeError:
-        raise TypeError('unhashable name attribute', name)
-    return name
-
-def shape_filter(array: np.ndarray) -> tp.Tuple[int, int]:
-    '''Represent a 1D array as a 2D array with length as rows of a single-column array.
-
-    Return:
-        row, column count for a block of ndim 1 or ndim 2.
-    '''
-    if array.ndim == 1:
-        return array.shape[0], 1
-    return array.shape #type: ignore
-
-def column_2d_filter(array: np.ndarray) -> np.ndarray:
-    '''Reshape a flat ndim 1 array into a 2D array with one columns and rows of length. This is used (a) for getting string representations and (b) for using np.concatenate and np binary operators on 1D arrays.
-    '''
-    # it is not clear when reshape is a copy or a view
-    if array.ndim == 1:
-        return np.reshape(array, (array.shape[0], 1))
-    return array
-
-def column_1d_filter(array: np.ndarray) -> np.ndarray:
-    '''
-    Ensure that a column that might be 2D or 1D is returned as a 1D array.
-    '''
-    if array.ndim == 2:
-        # could assert that array.shape[1] == 1, but this will raise if does not fit
-        return np.reshape(array, array.shape[0])
-    return array
-
-def row_1d_filter(array: np.ndarray) -> np.ndarray:
-    '''
-    Ensure that a row that might be 2D or 1D is returned as a 1D array.
-    '''
-    if array.ndim == 2:
-        # could assert that array.shape[0] == 1, but this will raise if does not fit
-        return np.reshape(array, array.shape[1])
-    return array
+from arraykit import mloc
```

Comment:
FYI, this is how you tell mypy to "reexport" an import:

```suggestion
from arraykit import mloc as mloc
```

Just adding "as" with the same name should do it.

I also just realized that we should probably add a `py.typed` file to ArrayKit, just as we did here.

Reply:
Thanks, that does the trick!